compiler/rustc_hir_analysis/src/collect/predicates_of.rs

```diff
@@ -452,7 +467,38 @@ pub(super) fn explicit_predicates_of<'tcx>(
                 //
                 // In the above code we want the anon const to have predicates in its param env for `T: Trait`
                 // and we would be calling `explicit_predicates_of(Foo)` here
-                return tcx.explicit_predicates_of(parent_def_id);
+                let parent_preds = tcx.explicit_predicates_of(parent_def_id);
+
+                // If we dont filter out `ConstArgHasType` predicates then every single defaulted const parameter
+                // will ICE because of #106994.
```

Comment:
This is really gross and makes me regret ever making #86580... on the bright side this added hack and the hack introduced by #86580 will be able to be completely removed in the near(?) future once we have `min_generic_const_exprs` which requires this PR to land

Reply:
Add "FIXME" or some other wording that indicates that this should be fixed?